### ISSUE TITLE
Enable swapping out the optimizer (e.g. AdamW instead of Adam)

### ISSUE THREAD

**Issue by jwuphysics** [maho3/ltu-ili#183]
Swapping out the optimizer can often leading to improved or more efficient optimization. [AdamW](https://docs.pytorch.org/docs/stable/generated/torch.optim.AdamW.html) is often a stronger default than Adam because it decouples the weight decay from the optimization step (see e.g. [arXiv:1711.05101](https://arxiv.org/abs/1711.05101)).

**Comment by jwuphysics** [comment]
@christopherlovell noted

> So for `sbi` I think you'd need to make a custom training loop https://sbi-dev.github.io/sbi/latest/tutorials/18_training_interface/#training-the-density-estimator

> For `lampe` it's as simple as swapping it out https://github.com/maho3/ltu-ili/blob/16d7c61a5d75125e7e1ba9188430bfdb06ea326d/ili/inference/runner_lampe.py#L246

> For pydelfi it's just a torch argument: https://pydelfi.readthedocs.io/en/latest/_modules/pydelfi/train.html#ConditionalTrainer.train

**Comment by maho3** [comment]
Formalizing this issue...

### Is your feature request related to a problem? Please describe.
Current optimizer (Adam) might not be optimal, potentially hindering optimization efficiency and performance.

### Describe the solution you'd like
Enable swapping out the optimizer, allowing use of alternatives like AdamW.

### Additional context
AdamW often performs better than Adam because it decouples the weight decay from the optimization step, potentially leading to more efficient optimization.

### Tasks
- [ ] Implement optimizer config for sbi
- [ ] Implement optimizer config for lampe
- [ ] Implement optimizer config for pydelfi
